### PR TITLE
Fix some site extension sdk bugs

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: src/Internal.AspNetCore.SiteExtension.Sdk/build/Internal.AspNetCore.SiteExtension.Sdk.targets [hunk 1]
```diff
@@ -32,7 +31,7 @@
       </ManifestFileContents>
     </PropertyGroup>
 
-    <ItemGroup Condition="'@(HostingStartupRuntimeStoreFrameworks->Count())' == '0'">
+    <ItemGroup Condition="'@(HostingStartupRuntimeStoreTargets->Count())' == '0'">
```
Comment (pakrym) [inline src/Internal.AspNetCore.SiteExtension.Sdk/build/Internal.AspNetCore.SiteExtension.Sdk.targets]: Wrong name here

File: src/Internal.AspNetCore.SiteExtension.Sdk/build/Internal.AspNetCore.SiteExtension.Sdk.targets [hunk 2]
```diff
@@ -23,7 +23,6 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Remove="Internal.AspNetCore.Sdk" />
-    <PackageReference Remove="Microsoft.NETCore.App" />
```
Comment (pakrym) [inline src/Internal.AspNetCore.SiteExtension.Sdk/build/Internal.AspNetCore.SiteExtension.Sdk.targets]: Something changed in SDK and we don't need this anymore

File: src/Internal.AspNetCore.SiteExtension.Sdk/build/Internal.AspNetCore.SiteExtension.Sdk.targets [hunk 3]
```diff
@@ -54,7 +53,7 @@
 
     <!-- Generate runtime store -->
     <WriteLinesToFile File="$(_RuntimeStoreManifestFile)" Lines="$(ManifestFileContents)" Overwrite="true" Encoding="Unicode"/>
-    <Exec Command="dotnet store -m $(_RuntimeStoreManifestFile) -f %(HostingStartupRuntimeStoreTargets.Identity) --framework-version $(HostingStartupRuntimeFrameworkVersion) -r %(HostingStartupRuntimeStoreTargets.Runtime) --output $(_RuntimeStoreOutput) --skip-optimization /p:DotNetRestoreSourcePropsPath=$(DotNetRestoreSourcePropsPath)" />
+    <Exec Command="dotnet store -m $(_RuntimeStoreManifestFile) -f %(HostingStartupRuntimeStoreTargets.Identity) --framework-version $(HostingStartupRuntimeFrameworkVersion) -r %(HostingStartupRuntimeStoreTargets.Runtime) --output $(_RuntimeStoreOutput)\%(HostingStartupRuntimeStoreTargets.Runtime) --skip-optimization /p:DotNetRestoreSourcePropsPath=$(DotNetRestoreSourcePropsPath)" />
```
Comment (pakrym) [inline src/Internal.AspNetCore.SiteExtension.Sdk/build/Internal.AspNetCore.SiteExtension.Sdk.targets]: Isolate runtime store builds from one another

File: src/Internal.AspNetCore.SiteExtension.Sdk/build/Internal.AspNetCore.SiteExtension.Sdk.targets (hunk 2 above)
Comment (natemcmaster) [reply src/Internal.AspNetCore.SiteExtension.Sdk/build/Internal.AspNetCore.SiteExtension.Sdk.targets]: Probably related to `FrameworkReference` changes.

**Comment by natemcmaster** [comment]
FYI - updates to BuildTools don't flow automatically anymore into aspnetcore. You'll need to make a PR to update BuildTools. The easiest way to update is:
```
build.cmd -u /t:noop
```

**Pull request merged**